client/src/routes/app/[country]/[map]/[year]/initialize/LoadRegions.ts

```diff
@@ -73,6 +78,28 @@ function lockRegion(regionID: string) {
 }
 
 function loadRegions(node: HTMLDivElement): void {
+	/*Load Candidates from SVG, Candidates are defined in <defs> using a <foreignObject> specifically named "region-candidates" that looks like the following:
+	<defs
+      	<foreignObject class="region-candidates">
+      		{"id":0, "name":"Joe Biden", "margins": [{ "color": "#FFF" }, { "color": "#000" }, { "color": "#333" }]}
+      		{"id":1, "name":"Donald Trump", "margins": [{ "color": "#FFF" }, { "color": "#000" }, { "color": "#333" }]}
+      	</foreignObject>
+   	</defs>
+	*/
+	const candidates = node.querySelector('.region-candidates')?.innerHTML;
+	const candidateStringArray = candidates?.split('\n') || []; //Seperate candidate object strings by newline character
+	const newCandidates: Candidate[] = [];
+	for (const candidateString of candidateStringArray) {
+		if (candidateString.trim() !== '') {
+			//If string has content
+			const newCandidate: Candidate = JSON.parse(candidateString);
+			newCandidates.push(newCandidate);
+		}
+	}
+	if (newCandidates.length !== 0) {
+		//If no candidates are defined in SVG, use generics defined in stores/Candidates.ts
+		CandidatesStore.set(newCandidates);
+	}
```

Comment:
candidateString was a bad variable name for that mapping.

Reply:
Also, what are your thoughts on using an attribute somewhere to store the JSON data in a string?

I think this would make it easier for people using an SVG editor to add the data. Because I'm not sure how valid it is to just store JSON data inside the foreignObjects element. Would something like inkscape or adobe even let you do that through their GUI?

![image](https://user-images.githubusercontent.com/13600696/209347704-9cb7b12b-816e-4036-a1f4-746e98ff49e2.png)

Reply:
Also, we could use Zod here to validate the structure of the data. And we can prevent someone from sending bad data/types to the system.

Reply:
Inkscape has a very feature-rich XML editor that allows you to set an attribute on the SVG like above, yes.

Going to look at using Zod.